libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt

```diff
@@ -181,11 +159,42 @@ internal class MapboxTripSession(
         if (state == TripSessionState.STARTED) {
             return
         }
+        setNavigatorObserver()
         tripService.startService()
         startLocationUpdates()
         state = TripSessionState.STARTED
     }
 
+    private fun setNavigatorObserver() {
+        navigator.setNavigatorObserver(
+            object : NavigatorObserver() {
+                override fun onStatus(origin: NavigationStatusOrigin, status: NavigationStatus) {
+                    mainJobController.scope.launch {
```

Comment:
Is this not invoked on the main thread? Why do we need to wrap contents in the coroutine?